### ISSUE TITLE
Exporting Saga and Sequence Diagram views

### ISSUE THREAD

**Issue by MpDzik** [Particular/ServiceInsight#768]
It would be great to be able to export the Saga and Sequence Diagram views to a bitmap or HTML file. Often I need to capture a sample saga for documentation purposes and currently I do it by taking a screenshot, however, when the view is too long to fit on one screen, then it is very tedious to take multiple screenshots and merge them manually.

**Comment by HEskandari** [comment]
Thanks for openining the issue @MpDzik. We've plans to support this when we released the feature but it felt a plan export to image would not be that useful as there's a lot of interactive information, e.g. in Sequence Diagram when you move your mouse on the endpoints/messagers which would be missing from the exported image.

We'll review this again soon and I'll keep you posted here.

**Comment by benagain** [comment]
> it felt a plan export to image would not be that useful as there's a lot of interactive information

I think it would be a useful feature nonetheless.  Just like @MpDzik I want to keep a record of the life of a given saga so that I can reference it in bug reports, or discuss it in slack channels, etc. 

Even without the dynamic features offered by the ServiceInsight application the saga view is very useful; it shows all the messages received and sent by the saga, with the message data; it shows the state changes the saga goes through over time.  Being able to put that into a document of some kind has a good deal of value.

**Comment by MpDzik** [comment]
> 
> 
> > it felt a plan export to image would not be that useful as there's a lot of interactive information
> 
> I think it would be a useful feature nonetheless. Just like @MpDzik I want to keep a record of the life of a given saga so that I can reference it in bug reports, or discuss it in slack channels, etc.
> 
> Even without the dynamic features offered by the ServiceInsight application the saga view is very useful; it shows all the messages received and sent by the saga, with the message data; it shows the state changes the saga goes through over time. Being able to put that into a document of some kind has a good deal of value.

I agree. Currently I just use screenshots to capture the saga sequence, but it is very problematic when the saga is too long to fit on a single screen. Then I need to grab a few screenshots and merge them manually. Being able to export the diagram to a PNG or JPG file would be awesome.
